src/TBuffer.cpp

```diff
@@ -787,10 +791,24 @@ void TBuffer::setBufferSize(int s, int batch)
     if (batch >= s) {
         batch = s / 10;
     }
-    mLinesLimit = s;
+    // clip the maximum to something reasonable, else users will abuse this, and then complain
+    mLinesLimit = std::min(s, getMaxBufferSize());
     mBatchDeleteSize = batch;
 }
 
+// naive calculation to get a reasonable limit for a maximum buffer size
+int TBuffer::getMaxBufferSize()
+{
+    const int64_t physicalMemoryTotal = mudlet::self()->getPhysicalMemoryTotal();
+    // Mudlet is 32bit mainly on Windows, see where the practical limit for a process 2GB:
+    // https://docs.microsoft.com/en-us/windows/win32/memory/memory-limits-for-windows-releases#memory-and-address-space-limits
+    // otherwise set to 80% of physical ram, ignoring that swap can exist
```

Comment:
Small improvement IMO:
```suggestion
    // set to 80% of what is available to us, ignoring that swap can exist
```